hudi-integ-test/src/main/java/org/apache/hudi/integ/testsuite/generator/GenericRecordFullPayloadGenerator.java

```diff
@@ -205,45 +193,36 @@ private Object typeConvert(Schema schema) {
       case LONG:
         return getNextConstrainedLong();
       case STRING:
-        return UUID.randomUUID().toString();
+       return UUID.randomUUID().toString();
       case ENUM:
-        List<String> enumSymbols = localSchema.getEnumSymbols();
-        return new GenericData.EnumSymbol(localSchema, enumSymbols.get(random.nextInt(enumSymbols.size() - 1)));
+        List<String> enumSymbols = fieldSchema.getEnumSymbols();
+        return new GenericData.EnumSymbol(fieldSchema, enumSymbols.get(random.nextInt(enumSymbols.size() - 1)));
       case RECORD:
-        return convert(localSchema);
+        return getNewPayload(fieldSchema);
       case ARRAY:
-        Schema elementSchema = localSchema.getElementType();
+        Schema.Field elementField = new Schema.Field(field.name(), fieldSchema.getElementType(), "", null);
         List listRes = new ArrayList();
-        if (isPrimitive(elementSchema) && this.shouldAddMore) {
-          int numEntriesToAdd = numEntriesToAdd(elementSchema);
-          while (numEntriesToAdd > 0) {
-            listRes.add(typeConvert(elementSchema));
-            numEntriesToAdd--;
-          }
-        } else {
-          listRes.add(typeConvert(elementSchema));
+        int numEntriesToAdd = extraEntriesMap.getOrDefault(field.name(), 1);
+        while (numEntriesToAdd > 0) {
```

Comment:
minor. we might as well do while (numEntriesToAdd-- > 0) { 
and remove line 208.

Reply:
Done